2020-10-06 15:13:08.146523: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
Running tests under Python 3.6.12: C:\ProgramData\Anaconda3\envs\keras\python.exe
[ RUN      ] OptimizerTest.testAggregationMethod
2020-10-06 15:13:12.017539: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library nvcuda.dll
2020-10-06 15:13:12.125311: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:12.127095: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:12.131445: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:12.134790: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:12.136412: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:12.141251: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:12.144440: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:12.153614: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:12.154546: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:12.155204: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2020-10-06 15:13:12.168795: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:12.169376: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:12.169716: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:12.170059: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:12.170398: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:12.170746: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:12.171091: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:12.171436: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:12.172270: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.110448: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.110863: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.111100: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.111836: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
WARNING:tensorflow:From C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\ops\resource_variable_ops.py:1630: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future version.
Instructions for updating:
If using Keras pass *_constraint arguments to layers.
W1006 15:13:13.124667 21292 deprecation.py:506] From C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\ops\resource_variable_ops.py:1630: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future version.
Instructions for updating:
If using Keras pass *_constraint arguments to layers.
[  FAILED  ] OptimizerTest.testAggregationMethod
[ RUN      ] OptimizerTest.testBasic
WARNING:tensorflow:From C:\ProgramData\Anaconda3\envs\keras\lib\contextlib.py:60: TensorFlowTestCase.test_session (from tensorflow.python.framework.test_util) is deprecated and will be removed in a future version.
Instructions for updating:
Use `self.session()` or `self.cached_session()` instead.
W1006 15:13:13.155905 21292 deprecation.py:323] From C:\ProgramData\Anaconda3\envs\keras\lib\contextlib.py:60: TensorFlowTestCase.test_session (from tensorflow.python.framework.test_util) is deprecated and will be removed in a future version.
Instructions for updating:
Use `self.session()` or `self.cached_session()` instead.
2020-10-06 15:13:13.163710: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.164236: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.164580: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.164963: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.165318: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.165660: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.166003: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.166345: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.166896: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.167200: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.167547: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.167774: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.168252: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
[  FAILED  ] OptimizerTest.testBasic
[ RUN      ] OptimizerTest.testComputeGradientsWithTensors
2020-10-06 15:13:13.634612: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.636014: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.636951: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.637883: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.638800: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.639490: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.640037: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.640592: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.641478: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.641951: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.642502: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.642845: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.643572: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
[  FAILED  ] OptimizerTest.testComputeGradientsWithTensors
[ RUN      ] OptimizerTest.testConstraint
2020-10-06 15:13:13.653805: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.654763: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.655405: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.656050: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.656679: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.657312: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.657953: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.658594: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.659628: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.660266: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.661008: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.661472: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.662406: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
[  FAILED  ] OptimizerTest.testConstraint
[ RUN      ] OptimizerTest.testGradientsAsVariables
2020-10-06 15:13:13.696809: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.697372: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.697739: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.698116: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.698477: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.698846: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.699215: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.699585: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.700178: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.700504: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.700878: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.701119: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.701629: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
[  FAILED  ] OptimizerTest.testGradientsAsVariables
[ RUN      ] OptimizerTest.testNoGradients
2020-10-06 15:13:13.746315: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.746884: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.747256: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.747628: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.747993: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.748371: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.748743: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.749114: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.749705: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.750027: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.750400: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.750632: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.751139: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
2020-10-06 15:13:13.778841: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.779393: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.779759: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.780122: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.780483: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.780908: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.781358: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.781795: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.782479: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.782799: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.783171: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.783388: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.783867: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
[       OK ] OptimizerTest.testNoGradients
[ RUN      ] OptimizerTest.testNoGradientsForAnyVariables_ApplyGradients
2020-10-06 15:13:13.827081: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.827902: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.828439: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.828976: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.829506: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.829870: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.830240: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.830584: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.831139: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.831437: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.831786: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.832005: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.832497: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
[  FAILED  ] OptimizerTest.testNoGradientsForAnyVariables_ApplyGradients
[ RUN      ] OptimizerTest.testNoGradientsForAnyVariables_Minimize
2020-10-06 15:13:13.840234: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.840746: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.841093: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.841433: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.841772: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.842119: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.842465: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.842810: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.843367: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.843670: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.844021: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.844240: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.844705: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
[       OK ] OptimizerTest.testNoGradientsForAnyVariables_Minimize
[ RUN      ] OptimizerTest.testNoVariables
2020-10-06 15:13:13.876801: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.877342: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.877682: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.878028: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.878366: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.878706: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.879050: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.879392: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.879939: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.880236: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.880583: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.880802: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.881274: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
[       OK ] OptimizerTest.testNoVariables
[ RUN      ] OptimizerTest.testPrecomputedGradient
2020-10-06 15:13:13.919123: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.919679: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.920020: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.920358: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.920694: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.921040: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.921393: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.921813: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.922403: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.922753: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.923101: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.923318: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.923789: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
[  FAILED  ] OptimizerTest.testPrecomputedGradient
[ RUN      ] OptimizerTest.testTrainOp
2020-10-06 15:13:13.952058: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: GeForce RTX 2080 Ti major: 7 minor: 5 memoryClockRate(GHz): 1.65
pciBusID: 0000:73:00.0
2020-10-06 15:13:13.952573: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
2020-10-06 15:13:13.952958: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
2020-10-06 15:13:13.953299: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cufft64_100.dll
2020-10-06 15:13:13.953637: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library curand64_100.dll
2020-10-06 15:13:13.953975: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusolver64_100.dll
2020-10-06 15:13:13.954319: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cusparse64_100.dll
2020-10-06 15:13:13.954666: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudnn64_7.dll
2020-10-06 15:13:13.955223: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2020-10-06 15:13:13.955525: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-10-06 15:13:13.955875: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2020-10-06 15:13:13.956092: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2020-10-06 15:13:13.956562: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8686 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080 Ti, pci bus id: 0000:73:00.0, compute capability: 7.5)
[  FAILED  ] OptimizerTest.testTrainOp
[ RUN      ] OptimizerTest.test_session
[  SKIPPED ] OptimizerTest.test_session
======================================================================
ERROR: testAggregationMethod (__main__.OptimizerTest)
OptimizerTest.testAggregationMethod
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\framework\test_util.py", line 1216, in decorated
    return f(self, *args, **kwargs)
  File "optimizer_test.py", line 157, in testAggregationMethod
    aggregation_method=gradients_util.AggregationMethod.
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 413, in minimize
    name=name)
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 595, in apply_gradients
    ([str(v) for _, v, _ in converted_grads_and_vars],))
ValueError: No gradients provided for any variable: ["<tf.Variable 'Variable:0' shape=(2,) dtype=float16>", "<tf.Variable 'Variable_1:0' shape=(2,) dtype=float16>"].

======================================================================
ERROR: testBasic (__main__.OptimizerTest)
OptimizerTest.testBasic
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\framework\test_util.py", line 1080, in decorated
    f(self, *args, **kwargs)
  File "optimizer_test.py", line 119, in testBasic
    opt_op = sgd_op.minimize(loss, global_step, [var0, var1])
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 413, in minimize
    name=name)
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 595, in apply_gradients
    ([str(v) for _, v, _ in converted_grads_and_vars],))
ValueError: No gradients provided for any variable: ["<tf.Variable 'a_0:0' shape=(2,) dtype=float16>", "<tf.Variable 'b_0:0' shape=(2,) dtype=float16>"].

======================================================================
ERROR: testComputeGradientsWithTensors (__main__.OptimizerTest)
OptimizerTest.testComputeGradientsWithTensors
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\framework\test_util.py", line 1080, in decorated
    f(self, *args, **kwargs)
  File "optimizer_test.py", line 467, in testComputeGradientsWithTensors
    sgd_op.apply_gradients(grads_and_vars)
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 595, in apply_gradients
    ([str(v) for _, v, _ in converted_grads_and_vars],))
ValueError: No gradients provided for any variable: ['Tensor("Const:0", shape=(), dtype=float32)'].

======================================================================
ERROR: testConstraint (__main__.OptimizerTest)
OptimizerTest.testConstraint
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\framework\test_util.py", line 1216, in decorated
    return f(self, *args, **kwargs)
  File "optimizer_test.py", line 521, in testConstraint
    opt_op = sgd_op.minimize(cost, global_step, [var0, var1])
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 413, in minimize
    name=name)
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 595, in apply_gradients
    ([str(v) for _, v, _ in converted_grads_and_vars],))
ValueError: No gradients provided for any variable: ["<tf.Variable 'Variable:0' shape=(2,) dtype=float32>", "<tf.Variable 'Variable_1:0' shape=(2,) dtype=float32>"].

======================================================================
ERROR: testGradientsAsVariables (__main__.OptimizerTest)
OptimizerTest.testGradientsAsVariables
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\framework\test_util.py", line 1080, in decorated
    f(self, *args, **kwargs)
  File "optimizer_test.py", line 427, in testGradientsAsVariables
    opt_op = sgd_op.apply_gradients(converted_grads_and_vars)
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 595, in apply_gradients
    ([str(v) for _, v, _ in converted_grads_and_vars],))
ValueError: No gradients provided for any variable: ["<tf.Variable 'a0:0' shape=(2,) dtype=float16>", "<tf.Variable 'b0:0' shape=(2,) dtype=float16>"].

======================================================================
ERROR: testPrecomputedGradient (__main__.OptimizerTest)
OptimizerTest.testPrecomputedGradient
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\framework\test_util.py", line 1216, in decorated
    return f(self, *args, **kwargs)
  File "optimizer_test.py", line 207, in testPrecomputedGradient
    cost, global_step, [var0, var1], grad_loss=grad_loss)
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 413, in minimize
    name=name)
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 595, in apply_gradients
    ([str(v) for _, v, _ in converted_grads_and_vars],))
ValueError: No gradients provided for any variable: ["<tf.Variable 'Variable:0' shape=(2,) dtype=float16>", "<tf.Variable 'Variable_1:0' shape=(2,) dtype=float16>"].

======================================================================
ERROR: testTrainOp (__main__.OptimizerTest)
OptimizerTest.testTrainOp
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\framework\test_util.py", line 1216, in decorated
    return f(self, *args, **kwargs)
  File "optimizer_test.py", line 489, in testTrainOp
    opt_op = sgd_op.minimize(cost, global_step, [var0, var1])
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 413, in minimize
    name=name)
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\training\optimizer.py", line 595, in apply_gradients
    ([str(v) for _, v, _ in converted_grads_and_vars],))
ValueError: No gradients provided for any variable: ["<tf.Variable 'Variable:0' shape=(2,) dtype=float32>", "<tf.Variable 'Variable_1:0' shape=(2,) dtype=float32>"].

======================================================================
FAIL: testNoGradientsForAnyVariables_ApplyGradients (__main__.OptimizerTest)
OptimizerTest.testNoGradientsForAnyVariables_ApplyGradients
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\keras\lib\site-packages\tensorflow_core\python\framework\test_util.py", line 1080, in decorated
    f(self, *args, **kwargs)
  File "optimizer_test.py", line 357, in testNoGradientsForAnyVariables_ApplyGradients
    sgd_op.apply_gradients([(None, var0), (None, var1)])
AssertionError: ValueError not raised

----------------------------------------------------------------------
Ran 12 tests in 2.031s

FAILED (failures=1, errors=7, skipped=1)
